hkmc2/shared/src/main/scala/hkmc2/syntax/Parser.scala

```diff
@@ -900,6 +900,12 @@ abstract class Parser(
       case (NEWLINE, _) :: (KEYWORD(kw), _) :: _
       if kw.canStartInfixOnNewLine && kw.leftPrecOrMin > prec
       && infixRules.kwAlts.contains(kw.name)
+      && kw != Keyword.`do` // This is to avoid the following case:
```

Comment:
`!=` should be avoided. 
```suggestion
      && kw isnt Keyword.`do` // This is to avoid the following case:
```